src/test/run-make-fulldeps/instrument-coverage-cov-reports-base/Makefile

```diff
@@ -0,0 +1,77 @@
+# needs-profiler-support
```

Comment:
> I've been seeing a number of makefiles go by with relatively similar contents

I don't understand the comment. Are you talking about Makefiles in other tests, not this PR?

This PR refactored the Makefiles so there is almost zero duplication, but it supports important test variations and combinations.

I think this is a good solution, given the existing test framework.

If this is a comment in generation, not specific to this PR, let me know.

Otherwise, I'm not sure what you want me to do (and not sure if it is in scope).

Reply:
I see now. I didn't notice that you deleted the other Makefile that was like this. I also haven't kept up with how many Makefile tests have been added, it just seems like I've reviewed a lot of them.

compiletest has Rust implementations for different kinds of tests, and I was asking what you thought about adding a new kind of test instead of doing it in Makefiles. Makefile tests are great for one-off tests but at some point it becomes more maintainable to implement their logic in Rust. I can't tell if we're nearing that point or not.

Definitely not something that needs to be done in this change (and sorry for not making that clear.) Especially given that there's little duplication, I'm fine with leaving as-is for now.

Reply:
Ack